network/dag/verifier.go

```diff
@@ -28,11 +32,20 @@ func NewTransactionSignatureVerifier(resolver types.KeyResolver) Verifier {
 			}
 		} else {
 			signingTime := tx.SigningTime()
-			pk, err := resolver.ResolvePublicKey(tx.SigningKeyID(), &signingTime)
-			if err != nil {
-				return fmt.Errorf("unable to verify transaction signature, can't resolve key (kid=%s): %w", tx.SigningKeyID(), err)
+			if signingTime.After(didDocumentResolveEPoch) {
+				pk, err := resolver.ResolvePublicKeyFromOriginatingTransaction(tx.SigningKeyID(), tx.Ref())
+				if err != nil {
+					return fmt.Errorf("unable to verify transaction signature, can't resolve key by TX ref (kid=%s, txhash=%s): %w", tx.SigningKeyID(), tx.Ref().String(), err)
+				}
+				signingKey = pk
+			} else {
+				// legacy resolving for older documents
+				pk, err := resolver.ResolvePublicKey(tx.SigningKeyID(), &signingTime)
```

Comment:
Nitpick: Wouldn't it be better to use `resolve.ResolvePublicKey` for the current implementation and something like: `resolver.ResolveTimeBasedPublicKey` for the legacy implementation?